### PR TITLE
V2 people picker revamp

### PR DESCRIPTION
**Pull request by praveen970**
Modified V2 people picker internal logic.
People picker now uses V2TextField
Removed leading and trailing accessory views, replaced by textfield icon views. (Breaking change)
Partially fixed accessibility
Note: Known issue (accessibility does not announce when keyboard is shown)

Modified V2TextField:
1. Textfield now has decorationBox parameter to modify the textbox UI
2. added contentDescription parameter to add any semantic info

### Validations
peer review 
auto tests

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] Automated Tests
- [x] Documentation and demo app examples
- [x] VoiceOver and Keyboard Accessibility
- [x] Internationalization and RTL layouts
- [x] Size classes and window sizes (notched devices, multitasking, different window sizes, etc)

### CODE REVIEW

File: fluentui_peoplepicker/src/main/java/com/microsoft/fluentui/tokenized/peoplepicker/PeoplePicker.kt [hunk 1]
```diff
@@ -98,86 +95,85 @@ fun PeoplePicker(
...
     searchHint: String? = null,
+    leadingRestIcon: ImageVector? = null,
+    leadingFocusIcon: ImageVector? = null,
+    leadingIconContentDescription: String? = null,
+    trailingAccessoryIcon: FluentIcon? = FluentIcon(
+        SearchBarIcons.Dismisscircle,
+        contentDescription = LocalContext.current.resources.getString(R.string.fluentui_clear_text)
+    ),
+    contentDescription: String? = null,
     peoplePickerTokens: PeoplePickerTokens? = null
```
Comment (mishramayank1) [inline fluentui_peoplepicker/src/main/java/com/microsoft/fluentui/tokenized/peoplepicker/PeoplePicker.kt:110]: pass the peoplePickerTokens to TextField

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/PeoplePickerTokens.kt [hunk 2]
```diff
@@ -15,102 +15,16 @@ import com.microsoft.fluentui.theme.token.FluentGlobalTokens
...
-
-    @Composable
-    open fun assistiveTextPadding(peoplePickerInfo: PeoplePickerInfo): PaddingValues {
-        return PaddingValues(
-            top = FluentGlobalTokens.size(FluentGlobalTokens.SizeTokens.Size40),
-            bottom = FluentGlobalTokens.size(FluentGlobalTokens.SizeTokens.Size40)
-        )
-    }
-
     @Composable
```
Comment (mishramayank1) [inline fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/PeoplePickerTokens.kt:28]: hintColor & hintTextTypography are also available in TexFieldTokens. These should come for TexFieldTokens unless required overriding.

File: FluentUI.Demo/src/androidTest/java/com/microsoft/fluentuidemo/demos/V2PeoplePickerUITest.kt [hunk 3]
```diff
@@ -53,7 +54,7 @@ class V2PeoplePickerUITest {
             })
         }
         composeTestRule.onRoot().printToLog("testPeoplePickerChip")
-        val textField = composeTestRule.onNodeWithTag("Text field")
```
Comment (mishramayank1) [inline FluentUI.Demo/src/androidTest/java/com/microsoft/fluentuidemo/demos/V2PeoplePickerUITest.kt:56]: localize the Tag

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2PeoplePickerActivity.kt [hunk 4]
```diff
@@ -176,30 +162,17 @@ class V2PeoplePickerActivity : V2DemoActivity() {
...
-                    },
-                    trailingAccessoryContent = {
-                        Icon(
-                            Icons.Filled.Add,
-                            contentDescription = "Add",
-                            modifier = Modifier
-                                .size(24.dp)
-                        )
-                    },
+                    contentDescription = "Total recipients: ${selectedPeopleList.size}",
```
Comment (mishramayank1) [inline FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2PeoplePickerActivity.kt:175]: Localize string

File: FluentUI.Demo/src/androidTest/java/com/microsoft/fluentuidemo/demos/V2PeoplePickerUITest.kt [hunk 5]
```diff
@@ -139,7 +140,7 @@ class V2PeoplePickerUITest {
                 }
             )
         }
-        val textField = composeTestRule.onNodeWithTag("Text field")
+        val textField = composeTestRule.onNodeWithTag("V2PeoplePicker")
```
Comment (mishramayank1) [inline FluentUI.Demo/src/androidTest/java/com/microsoft/fluentuidemo/demos/V2PeoplePickerUITest.kt:143]: localize the Tag

**Review by mishramayank1 — APPROVED** [review]

File: fluentui_peoplepicker/src/main/java/com/microsoft/fluentui/tokenized/peoplepicker/PeoplePicker.kt [hunk 6]
```diff
@@ -98,86 +95,85 @@ fun PeoplePicker(
...
     errorString: String? = null,
     searchHint: String? = null,
+    leadingRestIcon: ImageVector? = null,
+    leadingFocusIcon: ImageVector? = null,
+    leadingIconContentDescription: String? = null,
+    trailingAccessoryIcon: FluentIcon? = FluentIcon(
+        SearchBarIcons.Dismisscircle,
+        contentDescription = LocalContext.current.resources.getString(R.string.fluentui_clear_text)
+    ),
+    contentDescription: String? = null,
```
Comment (mishramayank1) [inline fluentui_peoplepicker/src/main/java/com/microsoft/fluentui/tokenized/peoplepicker/PeoplePicker.kt:109]: add peoplePicker as prefix

File: FluentUI.Demo/src/androidTest/java/com/microsoft/fluentuidemo/demos/V2PeoplePickerUITest.kt (hunk 5 above)
Comment (praveen970) [reply FluentUI.Demo/src/androidTest/java/com/microsoft/fluentuidemo/demos/V2PeoplePickerUITest.kt:143]: These strings are local to the test class and not used outside, should we localize these strings? I think these tests wont fail even of the language is set to non english

File: FluentUI.Demo/src/androidTest/java/com/microsoft/fluentuidemo/demos/V2PeoplePickerUITest.kt (hunk 3 above)
Comment (praveen970) [reply FluentUI.Demo/src/androidTest/java/com/microsoft/fluentuidemo/demos/V2PeoplePickerUITest.kt:56]: These strings are local to the test class and not used outside, should we localize these strings? I think these tests wont fail even of the language is set to non english

File: fluentui_peoplepicker/src/main/java/com/microsoft/fluentui/tokenized/peoplepicker/PeoplePicker.kt [hunk 7]
```diff
@@ -342,4 +283,9 @@ fun rememberPeoplePickerItemDataList(
     ) {
         initialValue
     }
-}
\ No newline at end of file
+}
+
```
Comment (mishramayank1) [inline fluentui_peoplepicker/src/main/java/com/microsoft/fluentui/tokenized/peoplepicker/PeoplePicker.kt:287]: remove extra line

**Review by mishramayank1 — APPROVED** [review]

**Pull request merged**